### ISSUE TITLE
search query should not strip spaces

### ISSUE THREAD

**Issue by dwelle** [t9md/atom-narrow#300]
Spaces are significant. Right now the workaround is to use regexp search and replace spaces with `\s`, which is not ideal.

Narrow `0.65.0`, Win7, Atom `1.26.1`

**Comment by t9md** [comment]
Yes, that is limitation, you cannot search space surrounded keyword.
Current query spec is here https://github.com/t9md/atom-narrow/wiki/Query.

Using space as query keyword **separator**, I cannot keep bare white-space intact.

My suggestion is use `\s`. if it's not sufficient, try `[\s]`.

**Comment by dwelle** [comment]
Maybe having an option so we can toggle behavior to tokenize `AND` operator on `&` instead of SPACE?

**Comment by t9md** [comment]
That's one idea, then we have to escape `&` when we want to search literal `&`.

Another idea I implemented experimentally but removed is to "set query end manually by special command" such as `narrow-ui:set-query-end`.
Let's say you have "foo bar .|  buz" text at prompt line and `|` here is your cursor(not literal pipe char).
Then you execute `narrow-ui:set-query-end` via some short-cut.
Then "foo bar ." is passed to ag as one query. then remaining "buz" is used to filter ag's result as always.

What do you think of this idea?

**Comment by dwelle** [comment]
Why not.. If I understand it correctly, to search for `foo bar baz` via AND, you'd place multiple cursors: `foo|bar|buz` and run `narrow-ui:set-query-end`?

Now, it depends on how much you use what. Can be a drag if user wants to use both `AND` operator and `space` in their regexes, on regular basis.

For me, I personally didn't know that `foo bar baz` does AND-search. I just quite heavily use `space` in my regex searches. But since learning about the AND-search behavior, I can imagine I'll use both AND-search and `space`-in-regexes, quite regularly -- and then  `narrow-ui:set-query-end` might become a bottleneck.

IMO, using `&` for AND-search and having to escape `\&` for literal `&` char isn't that crazy since people (only me?) don't to too many searches containing `&` char.

Either way, I'm not opposed to any solution that allows my beloved `space` to be added to regexes :)
